Sources/StreamChat/APIClient/Endpoints/MessageEndpoints.swift

```diff
@@ -94,6 +94,54 @@ extension Endpoint {
     }
 }
 
+// MARK: - Reminder Endpoints
+
+extension Endpoint {
+    // Creates or updates a reminder for a message
+    static func createReminder(messageId: MessageId, request: ReminderRequestBody) -> Endpoint<ReminderResponsePayload> {
+        .init(
+            path: .reminder(messageId),
+            method: .post,
+            queryItems: nil,
+            requiresConnectionId: false,
+            body: request
+        )
+    }
+    
+    // Updates an existing reminder for a message
+    static func updateReminder(messageId: MessageId, request: ReminderRequestBody) -> Endpoint<ReminderResponsePayload> {
+        .init(
+            path: .reminder(messageId),
+            method: .patch,
+            queryItems: nil,
+            requiresConnectionId: false,
+            body: request
+        )
+    }
+    
+    // Deletes a reminder for a message
+    static func deleteReminder(messageId: MessageId) -> Endpoint<EmptyResponse> {
+        .init(
+            path: .reminder(messageId),
+            method: .delete,
+            queryItems: nil,
+            requiresConnectionId: false,
+            body: nil
+        )
+    }
+    
+    // Queries reminders with the provided parameters
+    static func queryReminders(query: MessageReminderListQuery) -> Endpoint<RemindersQueryPayload> {
+        .init(
+            path: .reminders,
+            method: .post,
+            queryItems: nil,
+            requiresConnectionId: false,
```

Comment:
how do we receive reminder events if we don't require connection id?

Reply:
Yeah this might need to be true, will need to recheck

Reply:
Although, I think reminders will be probably receive events even if you don't query them. Since you might not have a reminders list. But still want to get reminded of your reminders. So this is not needed AFAIK

Reply:
ok, just checking - if it works without the connection id we can resolve this one.